ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor.hs

```diff
@@ -667,3 +766,31 @@ wakeupDecision st _now =
     decisionState = st,
     decisionJobs  = []
   }
+
+
+-- | A node is classified as a hidden relay or a BP if it is configured so that
+-- it can only have a chance to be connected to local roots. See `HiddenRelayOrBP`.
+--
+-- In the second case mentioned in `HiddenRelayOrBP` we are slightly more strict.
+-- We not only check the targets but also current `PeerSelectionView`.
+--
+readPromiscuousMode
+  :: MonadSTM m
+  => PeerSelectionInterfaces m
+  -> UseBootstrapPeers
+  -> PeerSelectionSetsWithSizes peeraddr
+  -> STM m Bool
+readPromiscuousMode
+  PeerSelectionInterfaces { readUseLedgerPeers }
+  useBootstrapPeers
+  PeerSelectionView {
+    viewKnownBootstrapPeers = (knownBootstrapPeers, _)
+  }
+  =
```

Comment:
`readPromiscuousMode` also must take peersharing into account.
One can run a node without any localroots, bootstrap peers, and ledger peers . And then only depend on peersharing.

Reply:
Done